resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/v2/ResourceStagingServiceRetrofit.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.rest.kubernetes.v2
+
+import okhttp3.{RequestBody, ResponseBody}
+import retrofit2.Call
+import retrofit2.http.{Multipart, Streaming}
+
+/**
+ * Retrofit-compatible variant of {@link ResourceStagingService}. For documentation on
+ * how to use this service, see the aforementioned JAX-RS based interface.
+ */
+private[spark] trait ResourceStagingServiceRetrofit {
+
+  @Multipart
+  @retrofit2.http.PUT("/api/resources/upload")
```

Comment:
this is just an early API, so I'd expect it to change somewhat going forward.  But we should also try at least somewhat to 

I'd be a fan of:
1) putting `v0/` in the URLs in the (probable) case this changes down the road
2) splitting the overloaded purpose of the secret into a separate resource ID and resource secret.  That way in the future if we need an ID for a highly-available version of this service, or want to log in various places the resource that's being downloaded/uploaded, or want to collect metrics on this service for upload/download counts on a per-resource basis, we don't have to use the sensitive secret as the resource identifier everywhere.

It's going to take some work to plumb the ID+secret everywhere vs just the secret that we use now, but I'm optimistic that it will be ultimately worth it for future flexibility.